### PR TITLE
DBT schema file generator - struct fields

### PR DESCRIPTION
**Pull request by Aruldharanikumar**
Refactored schema handling to support both normal and STRUCT fields

- Previously, only scalar fields were processed for policy tag mapping.
- Updated the script to recursively handle STRUCT (RECORD) fields, preserving their nested structure.
- Flattened nested fields into dot notation for better policy tag mapping and DBT export.
- Added logic to match policy tags for nested fields by traversing through subfields.
- Updated schema update logic to handle nested fields with proper `policyTags` association.
- Enhanced the `convert_schema_field_to_dict` function to process both scalar and nested fields.
- The new implementation ensures that policy tags are applied to both simple and nested data types.

### CODE REVIEW

**Review by duygugencc — APPROVED** [review]
LGTM!

File: prod/policy_tag_dbt_generator_service/dbt_schema_file_generator.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        updated_schemas = {}
+
+        for table_name, schema in schemas.items():
+            updated_fields = self.match_policy_tags_to_fields(schema, policy_tag_mapping)
+            updated_schemas[table_name] = updated_fields
+
+        return updated_schemas
+
+    def convert_to_dbt_format(self, updated_schemas):
+        """Convert updated schema JSON to DBT YAML format, only including fields with policy tags."""
```
Comment (Adamsven) [inline prod/policy_tag_dbt_generator_service/dbt_schema_file_generator.py:137]: If a sensitive field is in a struct or array, will this be able to return the correct dbt yaml for that?

**Review by Adamsven — APPROVED** [review]
LGTM

File: prod/policy_tag_dbt_generator_service/dbt_schema_file_generator.py (hunk 1 above)
Comment (Aruldharanikumar) [reply prod/policy_tag_dbt_generator_service/dbt_schema_file_generator.py:137]: yes!

**Pull request merged**